File: rsexamples-main/dotnet/RSExampleTest/AngleTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;

using RSExample.Math;

namespace RSExampleTest
{
    [TestClass]
    public class AngleTest
    {
        void AssertEqual(float a, float b)
        {
            Assert.AreEqual(a, b, 0.00001f);
        }

        [TestMethod]
        public void TestDegreesToRadians()
        {
            AssertEqual(MathF.PI, Angle.DegreesToRadians(180));
        }

        [TestMethod]
        public void TestEquality()
        {
            AssertEqual(Angle.Degrees(0), Angle.Degrees(360));
            AssertEqual(Angle.Degrees(180), Angle.Degrees(-180));
            AssertEqual(Angle.Degrees(-90), Angle.Degrees(270));
        }


        [TestMethod]
        public void TestOpposite()
        {
            AssertEqual(Angle.Degrees(-90), Angle.Opposite(Angle.Degrees(90)));
            AssertEqual(Angle.Degrees(270), Angle.Opposite(Angle.Degrees(90)));
        }

        [TestMethod]
        public void TestDiff()
        {
            AssertEqual(Angle.Degrees(0), Angle.Diff(Angle.Degrees(0), Angle.Degrees(360)));
            AssertEqual(Angle.Degrees(90), Angle.Diff(Angle.Degrees(45), Angle.Degrees(-45)));
            AssertEqual(Angle.Degrees(90), Angle.Diff(Angle.Degrees(-45), Angle.Degrees(45)));
            AssertEqual(Angle.Degrees(90), Angle.Diff(11.780972450961723f, -5.497787143782138f));
        }

        [TestMethod]
        public void TestDiffClockwise()
        {
            AssertEqual(Angle.Degrees(90), Angle.DiffClockwise(Angle.Degrees(45), Angle.Degrees(-45)));
            AssertEqual(Angle.Degrees(90), Angle.DiffClockwise(Angle.Degrees(45), Angle.Degrees(315)));
            AssertEqual(Angle.Degrees(270), Angle.DiffClockwise(11.780972450961723f, -5.497787143782138f));
        }

        [TestMethod]
        public void TestDiffCounterclockwise()
        {
            AssertEqual(Angle.Degrees(270), Angle.DiffCounterclockwise(Angle.Degrees(45), Angle.Degrees(-45)));
            AssertEqual(Angle.Degrees(270), Angle.DiffCounterclockwise(Angle.Degrees(45), Angle.Degrees(315)));
            AssertEqual(Angle.Degrees(90), Angle.DiffCounterclockwise(11.780972450961723f, -5.497787143782138f));
        }
    }
}

File: rsexamples-main/nodejs/test/angle_test.js
var assert = require('assert');
var angle = require("../src/math/angle");
var { close } = require("./utils");

describe("degrees-to-radians", function () {
  it("d2r", function () {
    assert.equal(Math.PI, angle.d2r(180));
  });
});

describe("equality", function () {
  it("0 == 360", function() {
    assert.equal(angle.degrees(0), angle.degrees(360));
  });
  it("180 == -180", function() {
    assert.equal(angle.degrees(180), angle.degrees(-180));
  });
  it("-90 == 270", function() {
    assert.equal(angle.degrees(-90), angle.degrees(270));
  });
});

describe("opposite", function () {
  it("opposite(90) == -90", function() {
    assert.equal(angle.opposite(angle.degrees(90)), angle.degrees(-90));
  });
  it("opposite(90) == 270", function() {    
    assert.equal(angle.opposite(angle.degrees(90)), angle.degrees(270));
  });
});

describe("diff", function () {
  it("0 == diff(0, 360)", function () {
    assert(close(angle.degrees(0), 
                 angle.diff(angle.degrees(0),
                            angle.degrees(360)),
                 0.00001));
  });
  it("90 == diff(45, -45)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diff(angle.degrees(45),
                            angle.degrees(-45)),
                 0.00001));
  });
  it("90 == diff(-45, 45)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diff(angle.degrees(-45),
                            angle.degrees(45)),
                 0.00001));
  });
  it("90 == diff(11.78rad, -5.49rad)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diff(11.780972450961723,
                            -5.497787143782138),
                 0.00001));
  });
});

describe("diffClockwise", function () {
  it("90 == diff(45, -45)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diffClockwise(angle.degrees(45),
                                     angle.degrees(-45)),
                 0.00001));
  });
  it("90 == diff(45, 315)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diffClockwise(angle.degrees(45),
                                     angle.degrees(315)),
                 0.00001));
  });
  it("270 == diff(11.78rad, -5.49rad)", function () {    
    assert(close(angle.degrees(270), 
                 angle.diffClockwise(11.780972450961723,
                                     -5.497787143782138),
                 0.00001));
  });
});

describe("diffCounterclockwise", function () {
  it("270 == diff(45, -45)", function () {    
    assert(close(angle.degrees(270), 
                 angle.diffCounterclockwise(angle.degrees(45),
                                            angle.degrees(-45)),
                 0.00001));
  });
  it("270 == diff(45, 315)", function () {    
    assert(close(angle.degrees(270), 
                 angle.diffCounterclockwise(angle.degrees(45),
                                            angle.degrees(315)),
                 0.00001));
  });
  it("90 == diff(11.78rad, -5.49rad)", function () {    
    assert(close(angle.degrees(90), 
                 angle.diffCounterclockwise(11.780972450961723,
                                            -5.497787143782138),
                 0.00001));
  });
});